### PR TITLE
Fix enum preselection in VB collection initializer

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/3133

### CODE REVIEW

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs [hunk 1]
```diff
@@ -2239,22 +2239,6 @@ private IEnumerable<TypeInferenceInfo> InferTypeInWhileStatement(WhileStatementS
 
                 return CreateResult(SpecialType.System_Boolean);
             }
-
-            private IEnumerable<TypeInferenceInfo> GetCollectionElementType(INamedTypeSymbol type)
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs]: moved up to base type so we can use from VB and C#

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/VisualBasic/LanguageServices/VisualBasicTypeInferenceService.TypeInferrer.vb [hunk 2]
```diff
@@ -999,35 +1000,44 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
...
 
                 ' New List(Of T) From { x }
                 If expression IsNot Nothing Then
-                    Dim expressionAddMethodSymbols = SemanticModel.GetCollectionInitializerSymbolInfo(expression).GetAllSymbols()
-                    Dim expressionAddMethodParameterTypes = expressionAddMethodSymbols _
-                        .Where(Function(a) DirectCast(a, IMethodSymbol).Parameters.Length = 1) _
-                        .Select(Function(a) New TypeInferenceInfo(DirectCast(a, IMethodSymbol).Parameters(0).Type))
+                    Dim expressionAddMethodSymbols = SemanticModel.GetCollectionInitializerSymbolInfo(expression).GetAllSymbols().OfType(Of IMethodSymbol)
+                    Dim expressionAddMethodParameterTypes = expressionAddMethodSymbols.
+                        Where(Function(m) m.Parameters.Length = 1).
```
Comment (dpoeschl) [inline src/Workspaces/SharedUtilitiesAndExtensions/Workspace/VisualBasic/LanguageServices/VisualBasicTypeInferenceService.TypeInferrer.vb]: Dots on the right?!?!!!!!

File: src/EditorFeatures/VisualBasicTest/Completion/CompletionProviders/EnumCompletionProviderTests.vb [hunk 3]
```diff
@@ -480,8 +484,59 @@ End Module</Text>.Value
...
+        Public Async Function TestInCollectionInitializer3() As Task
+            Dim markup = <Text><![CDATA[
+Imports System
+Imports System.Collections.Generic
+
+Class C
+    Sub Main()
+        Dim y = New List(Of DayOfWeek) From {
+            DayOfWeek.Monday,
+            $$
```
[the remark this answers is not in the document]
Comment (CyrusNajmabadi) [reply src/EditorFeatures/VisualBasicTest/Completion/CompletionProviders/EnumCompletionProviderTests.vb]: > What's the difference between Test2 and Test3? Appears to just be the newline?

Yup!

>If that makes a difference, should we have a test with everything on one line?

Yup!

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/VisualBasic/LanguageServices/VisualBasicTypeInferenceService.TypeInferrer.vb (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/SharedUtilitiesAndExtensions/Workspace/VisualBasic/LanguageServices/VisualBasicTypeInferenceService.TypeInferrer.vb]: Always :)

**Comment by CyrusNajmabadi** [comment]
Thanks!

**Pull request merged**